pkg/aggregator/aggregator.go

```diff
@@ -291,15 +291,29 @@ func mergeSpecs(dest, source *spec.Swagger, renameModelConflicts, ignorePathConf
 			from, to string
 		}
 		renames := []Rename{}
+
+	OUTERLOOP:
 		for k, v := range source.Definitions {
 			if usedNames[k] {
 				v2, found := dest.Definitions[k]
-				// Reuse model iff they are exactly the same.
+				// Reuse model if they are exactly the same.
 				if found && reflect.DeepEqual(v, v2) {
 					continue
 				}
-				i := 2
-				newName := fmt.Sprintf("%s_v%d", k, i)
+
+				// Reuse previously renamed model if one exists
+				var newName string
+				i := 1
+				for found {
```

Comment:
should you set initial value of found to `true` here?

Reply:
It's always `true` anyway, because for it to be `false`, `usedNames` would have to contain `k`, whereas `dest.Definitions` wouldn't. And that's never the case, because `usedNames` is initialized with all the keys from `dest.Definitions`.